src/sm/sm-eac.c

```diff
@@ -865,7 +865,7 @@ static int eac_gen_auth_4_mutual_authentication(sc_card_t *card,
 	EAC_GEN_AUTH_PACE_R *r_data = NULL;
 	unsigned char *d = NULL, *p;
 	int r, l;
-	unsigned char resp[SC_MAX_EXT_APDU_RESP_SIZE];
+	unsigned char resp[card->max_recv_size];
```

Comment:
you're right to shrink the buffers here. However, the rest of the code doesn't use variable length arrays, please stick to this.

Reply:
Will it be safe to reduce this buffer size world wide? (Polish card can not handle extended APDU-s during pace. It responds with part of the request in such case)